js/reference.js

```diff
@@ -2076,39 +2076,58 @@
         },
 
         /**
-         * If the current reference is derived from an association related table,
-         * this function will return a reference to the corresponding
-         * entity set from the corresponding association table denoted by the list of tuples.
+         * If the current reference is derived from an association related table, this function will delete the set of
+         * tuples included and return a set of success responses and a set of errors for the corresponding delete
+         * actions for the provided entity set from the corresponding association table denoted by the list of tuples.
          *
          * For example, assume
          * Table1(K1,C1) <- AssociationTable(FK1, FK2) -> Table2(K2,C2)
          * and the current tuples are from Table2 with k2 = "2" and k2 = "3".
-         * With origFKRData = {"k1": "1"} this function will return a reference
-         * to AssocitaitonTable with FK1 = "1" as a part of the path and FK2 = "2" and FK2 = "3"
+         * With origFKRData = {"k1": "1"} this function will return a set of success and error responses for
+         * delete requests to AssociattionTable with FK1 = "1" as a part of the path and FK2 = "2" and FK2 = "3"
          * as the filters that define the set and how they are related to Table1.
          *
-         * @param {Array} tuples - an array of ERMrest.Tuple objects from Table2 (from example above)
+         * To make sure a deletion occurs only for the tuples specified, we need to verify each reference path that
+         * is created includes a parent constraint and has one or more filters based on the other side of the association
+         * table's uniqueness constraint. Some more information about the validations that need to occur based on the above example:
+         *  - parent value has to be not null
+         *    - FK1 has to have a not null constraint
+         *  - child values have to have at least 1 value and all not null
+         *    - for FK2, all selected values are not null
+         *
+         * @param {Array} mainTuple - an ERMrest.Tuple from Table1 (from example above)
+         * @param {Array} tuples - an array of ERMrest.Tuple objects from Table2 (same as self) (from example above)
          *
-         * @returns {Array} an array of ERMrest.Reference for the AssociationTable that maps the supplied tuples to a row from Table1
+         * @returns {Object} an ERMrest.BatchUnlinkResponse "error" object
          **/
-        getBatchAssociationRef: function (tuples) {
-            var self = this;
+        deleteBatchAssociationRef: function (parentTuple, tuples) {
```

Comment:
Just like the `delete` function, this should receive `contextHeaderParams` that will be passed with all the delete requests that it will generate.